File: 05-null/src/main.c
#include <stdio.h>
#include <stdbool.h>

const char *get_string(bool flag) {
    if (flag) {
	return "all your base are belong to us";
    } else {
	return NULL;
    }
}

void print_string(const char *s) {
    if (s == NULL) {
	printf("Error: string is NULL!\n");
    } else {
	printf("%s\n", s);
    }
}

int main(void) {
    print_string(get_string(false));
    print_string(get_string(true));

    return 0;
}

File: 05-null/src/main.rs
fn get_string(flag: bool) -> Option<&'static str> {
    if flag {
	Some("All your base are belong to us")
    } else {
	None
    }
}

fn print_string(s: Option<&str>) {
    match s {
        Some(s) => println!("{}", s),
        None => println!("Error: string is None!"),
    }
}

fn main() {
    print_string(get_string(false));
    print_string(get_string(true));
}
